exercises/concept/meteorology/meteorology.go

```diff
@@ -0,0 +1,35 @@
+package meteorology
+
+// Please add suitable String functions to all types
+
+type TemperatureUnit int
+
+const (
+	Celsius    TemperatureUnit = 0
+	Fahrenheit TemperatureUnit = 1
+)
+
+type Temperature struct {
+	degree int
+	unit   TemperatureUnit
+}
+
```

Comment:
Wondering if we could: 
- Let the student implement the `Temperature`, `Speed` and `MetData` structs and provide instructions on what the fields should look like and comment in `meteorology.go` (in fact, the instructions so far are quite self-explanatory of how the structs should look like but we could tell the students what field names we expect in the test cases).
- Provide instructions on how the formatted strings should look like for each struct that implements `String` function.
- Give the following stubs and let them implement the functions:
```go
func (unit TemperatureUnit) String() string {
	panic("Please implement the String function for TempereatureUnit")
}

func (t Temperature) String() string {
	panic("Please implement the String function for Tempereature")
}

func (unit SpeedUnit) String() string {
	panic("Please implement the String function for SpeedUnit")
}

func (sp Speed) String() string {
	panic("Please implement the String function for Speed")
}

func (m MetData) String() string {
	panic("Please implement the String function for MetData")
}
```
This seems to me to resemble existing exercises (the limited ones which I've seen).

Reply:
Thanks for the review! 

I would prefer to give students the types of those structs in order to keep the focus on (implementing) the `Stringer` interface. I expect some students will not be used to the "enum type" pattern, so looking that up (and understanding it) will  be enough of a "distraction" from the core `Stringer` part. 

I left out the `String` methods because I would like students to think about their type. I expect that some students will write these as functions rather then methods. In other words, getting the signature of String() right is part of the intended learning outcome. 

I don't have particular views regarding the provision of more details of the expected string format. Did you have anything particular in mind?

Reply:
This is a tricky one. It is true that we usually provide the stubs for functions and methods, but like norbs says, in this case, writing the methods is part of the exercise itself. I don't expect this writing of the methods to be too hard, we have the interface concept before, and in the examples for this exercise, we also show how to implement the interface and write the `String` method.

So I don't think that including the stubs or not really matters that much in the end. We can start by not providing it and see if students struggle with it and change later if needed.